lib/beaker-pe/install/pe_utils.rb

```diff
@@ -443,17 +444,20 @@ def do_install hosts, opts = {}
             # Wait for PuppetDB to be totally up and running (post 3.0 version of pe only)
             sleep_until_puppetdb_started(database) unless pre30database
 
-            # Run the agent once to ensure everything is in the dashboard
-            install_hosts.each do |host|
-              on host, puppet_agent('-t'), :acceptable_exit_codes => [0,2]
-
-              # Workaround for PE-1105 when deploying 3.0.0
-              # The installer did not respect our database host answers in 3.0.0,
-              # and would cause puppetdb to be bounced by the agent run. By sleeping
-              # again here, we ensure that if that bounce happens during an upgrade
-              # test we won't fail early in the install process.
-              if host['pe_ver'] == '3.0.0' and host == database
-                sleep_until_puppetdb_started(database)
+            step "First puppet agent run" do
+              # Run the agent once to ensure everything is in the dashboard
+              install_hosts.each do |host|
+                on host, puppet_agent('-t'), :acceptable_exit_codes => [0,2]
+
+                # Workaround for PE-1105 when deploying 3.0.0
+                # The installer did not respect our database host answers in 3.0.0,
+                # and would cause puppetdb to be bounced by the agent run. By sleeping
+                # again here, we ensure that if that bounce happens during an upgrade
+                # test we won't fail early in the install process.
+                if host == database
+                  sleep_until_puppetdb_started(database)
+                  check_puppetdb_status_endpoint(database)
```

Comment:
Britt and I are discussing that we probably need to gate this function only to run on 2016.1.x, since the status end point won't be there for all versions.